source/Account.cpp

```diff
@@ -115,29 +113,87 @@ void Account::Save(DataWriter &out) const
 
 
 
-// How much the player currently has in the bank.
 int64_t Account::Credits() const
 {
 	return credits;
 }
 
 
 
-// Give the player credits (or pass  negative number to subtract). If subtracting,
-// the calling function needs to check that this will not result in negative credits.
+void Account::SetCredits(int64_t value)
+{
+	credits = value;
+}
+
+
+
 void Account::AddCredits(int64_t value)
 {
 	credits += value;
 }
 
 
 
-// Pay down extra principal on a mortgage.
+int Account::CreditScore() const
+{
+	return creditScore;
+}
+
+
+
+const vector<int64_t> &Account::History() const
+{
+	return history;
+}
+
+
+
+int64_t Account::NetWorth() const
+{
+	return history.empty() ? 0 : history.back();
+}
+
+
+
+void Account::AddHistory(int64_t amount) {
+	history.push_back(amount);
+	if(history.size() > HISTORY)
+		history.erase(history.begin());
+}
+
+
+
+const vector<Mortgage> &Account::Mortgages() const
+{
+	return mortgages;
+}
+
+
+
+void Account::AddMortgage(int64_t principal)
+{
+	mortgages.emplace_back(principal, creditScore);
+	credits += principal;
+}
+
+
+
+// Add a mortgage with an interest rate of zero and a short term.
+void Account::AddFine(int64_t amount)
+{
+	mortgages.emplace_back(amount, 0, 60);
+}
+
+
+
 void Account::PayExtra(int mortgage, int64_t amount)
 {
-	if(static_cast<unsigned>(mortgage) >= mortgages.size() || amount > credits
-			|| amount > mortgages[mortgage].Principal())
+	if(static_cast<unsigned>(mortgage) >= mortgages.size()
+		|| amount > credits
+		|| amount > mortgages[mortgage].Principal())
+	{
```

Comment:
```suggestion
```

Reply:
@quyykk I disagree with removing the curly braces here. Thoughts?

Reply:
b- but less line

Reply:
Lmao, but it look more gross your way